lib/runtime/storage/trie.go

```diff
@@ -178,6 +180,43 @@ func (s *TrieState) DeleteChild(key []byte) {
 	s.t.DeleteChild(key)
 }
 
+// DeleteChildLimit deletes up to limit of database entries by lexicographic order, return number
+//  deleted, true if all delete atherwise false
+func (s *TrieState) DeleteChildLimit(key []byte, limit *optional.Bytes) (uint32, bool, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	tr, err := s.t.GetChild(key)
+	if err != nil {
+		return 0, false, err
+	}
+	qtyEntries := uint32(len(tr.Entries()))
+	if limit == nil || !limit.Exists() {
+		s.t.DeleteChild(key)
```

Comment:
I think this should be `DeleteFromChild`, after firstly getting the list of keys in the child